### ISSUE TITLE
Update Scintilla to v3.10.4 or latter to fix accessibility issues on Linux

### ISSUE THREAD

**Issue by Nardol** [fantaisie-software/purebasic#70]
On Linux, Scintilla version provided by PureBasic is not accessible with Orca, the screen reader on Linux.
We can no read the content at all, by line or by characters all arrows don't produce any announcement 
The one used by Geany for example is fully accesssible.
Would it be possible to use the same Scintilla version for the PureBasic IDE and so make it accessible for Linux screen reader users?

**Comment by tajmone** [comment]
It seems that Geany uses Scintilla v3.10.4:

https://www.geany.org/documentation/releasenotes/

Last time I checked, PB was using Scintilla v3.4.2.

PS: Regarding Scintilla 4, see #59.

**Comment by Nardol** [comment]
> It seems that Geany uses Scintilla v3.10.4:
> 
> https://www.geany.org/documentation/releasenotes/
> 
In this case, IMHO using Scintilla v3.10.4 instead of the latest which seems to cause some problems could be a good way to fix this accessibility issue.

**Comment by tajmone** [comment]
I'm all-in for updating the third party tools used by the PureLibs to their latest version.

Ideally, all third party components should be updated with their latest version at each new PB version. But I guess this is a lot of work for a small team, in terms of building and testing, but it would be nice to see them updated at least once a year — some components haven't been updated for years, which means that we don't benefit from many bug fixes, new features, and that we might face problems when referring to their official documentation or tutorials (when the version gap is big).

**Comment by Nardol** [comment]
Idealy, sure components should be updated to their latest version. But it looks loke in this case the latest version make a lot of problem and even if accessibility is not the biggest issue for most people, it is a big one for some who simply cannot use the PB IDE att all when we cannot read anything inside the editor.
That's why I talk about a compromise, not the absolute ideal solution.

I hop this situation could be improved, as it is a recurrent issue reported by others users, by mail for example. Work have been mede to improve it in the component itself and staying with V3 major version, I hop, should not cause issues for PureBasic itself as it is a minor version update.

**Comment by tajmone** [comment]
> even if accessibility is not the biggest issue for most people, it is a big one for some who simply cannot use the PB IDE att all when we cannot read anything inside the editor.

Accessibility is (or should be) _always_ a major concern. Workers with disabilities have a right to accessible software tool, and can legally force their employer to switch to a software which accessible, in case the company provides its workers with non-accessible tool.

It's a matter of equal rights. Fortunately, there are laws to enforce these rights on people who don't spontaneously care about accessibility needs. A color blind worker has the right to claim a software that supports accessible color schemes for the UI, or that can at least be manually configured according to needs; yet, it's rare to find GUI tools that natively offer alternative UI schemes for the various types of color blindnesses (but good tools do have them). 

Software developers should be more aware above the impact that accessibility can have on their software products.

**Comment by fantaisie-software** [comment]
I will try to update it to lastest 3.x version and see if it works.

**Comment by b4n** [comment]
You need Scintilla 3.7.1 at the very minimum, but I'd recommend at least 3.7.4, and as new as you can get.  4.1.5 has important fixes, I'm not sure to which 3.x versions it maps, but Geany's version contains it all, so that should be fine.

**Comment by Nardol** [comment]
Any news about this issue?

**Comment by Nardol** [comment]
I see Scintilla has been updated some times ago, found it by random today by reading the changelog.
Now I can read code in the IDE, very nice!

I noticed some other issues (particularly keyboard navigation E.G. in the preferences), should I report these in separate issues or also here?

**Comment by alphasnd** [comment]
Done in 6.0x
